python/tank/deploy/descriptor.py

```diff
@@ -14,14 +14,99 @@
 
 import os
 import copy
+import sys
 
 from tank_vendor import yaml
 
 from .. import hook
 from ..util import shotgun
-from ..errors import TankError
+from ..errors import TankError, TankFileDoesNotExistError
 from ..platform import constants
 
+class PathedSingletonDescriptor(type):
+    """
+    Singleton metaclass for the descriptor classes that operate off of disk paths.
+
+    Each descriptor object is a singleton based on its install root path.
+    """
+    _instances = dict()
+
+    def __call__(cls, pc_path, bundle_install_path, location_dict, *args, **kwargs):
+        # Platform specific location support.
+        system = sys.platform
+        platform_keys = dict(
+            linux2="linux_path",
+            darwin="mac_path",
+            win32="windows_path",
+        )
+
+        platform_key = platform_keys.get(system, "unsupported_os")
+
+        if platform_key not in location_dict and "path" in location_dict:
+            dev_path = location_dict.get("path")
+        elif platform_key and platform_key in location_dict:
+            dev_path = location_dict.get(platform_key)
+        else:
+            dev_path = None
```

Comment:
We can actually ditch this metaclass entirely since we're going to skip treating dev descriptors as singletons (as discussed in other comments here). Problem solved!